File: SiteVisitsNetCore2020Web/ClientApp/src/app/models/IpAddress.ts
import { City } from './City';
import { Country } from './Country';
import { Region } from './Region';
import { Organization } from './Organization';
import { Visitor } from './Visitor';

export class IpAddress {    

    ipV4Address: string;
    numberOfTimes: number;
    postalCode: string;
    city: City;
    region: Region;
    country: Country;
  organization: Organization;
  visitor: Visitor

  constructor(options: {
    ipV4Address: string,
    numberOfTimes: number,
    postalCode: string,
    city: City,
    region: Region,
    country: Country,
    organization: Organization,
    visitor: Visitor
  }) {
    this.ipV4Address = options.ipV4Address;
    this.numberOfTimes = options.numberOfTimes;
    this.postalCode = options.postalCode;
    this.city = new City(options.city);
    this.region = new Region(options.region);
    this.country = new Country(options.country);
    this.organization = new Organization(options.organization);
    this.visitor = new Visitor(options.visitor);
  }
}

File: SiteVisitsNetCore2020Web/Models/IpAddress.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel.DataAnnotations;

namespace SiteVisitsNetCore2020Web.Models
{
    public class IpAddress
    {
        [Key]
        public string IpV4Address { get; set; }
        public int NumberOfTimes { get; set; }
        public string PostalCode { get; set; }
        public virtual City City { get; set; }
        public virtual Region Region { get; set; }
        public virtual Country Country { get; set; }
        public virtual Isp Isp { get; set; }
        public virtual Visitor Visitor { get; set; }
        public virtual List<IpAddressRelation> IpAddressRelations { get; set; }
    }
}
